tools/clang/test/CodeGenHLSL/unsignedShortHandMatrixVector.hlsl

```diff
@@ -0,0 +1,12 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
+
+// CHECK: uint2
+// CHECK: uint4x4
+
+unsigned int2 a;
+unsigned int4x4 b;
```

Comment:
Test for signed as well?  Also, try a all legal component types to ensure they translate (or pass through) properly.